[8.0.0-alpha] - Initial Settings are missing after fresh installation

### ⚠️ This issue respects the following points: ⚠️

- [x] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [x] This issue is **not** already reported on Github _(I've searched it)_.
- [x] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### What went wrong, what did you observe?

After I've installed the polls app in a dev environment, there is no button to create a new poll (see screenshot)

![Image](https://github.com/user-attachments/assets/d3f07b9e-2963-4b0c-b9c0-4e18bbad068c)


### What did you expect, how polls should behave instead?

I think it makes sense that you can create a poll in the Polls app. Accordingly, it should have a button for this 😀 .

### What steps does it need to replay this bug?

1. open polls app
2. click to button "Create poll" (or some other text)


### Installation method

Installed/updated from the appstore (Apps section of your site)

### Installation type

First time installation

### Affected polls version

 Version 8.0.0-alpha8

### Which browser did you use, when experiencing the bug?

- [x] Firefox
- [ ] Chrome
- [ ] Chromium/Chromium based (i.e. Edge)
- [ ] Safari
- [ ] Other/Don't know

### Other browser

_No response_

### Add your browser log here

jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. globals.js:36:27
$ is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. globals.js:36:27
[DEBUG] core: Unified search initialized! 
Object { app: "core", uid: "admin", level: 0 }
index.mjs:45:16
jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. 4 globals.js:36:27
XHRGET
https://devserver.nc.xxx/index.php/apps/polls/session?time=1729797613838
[HTTP/2 500  265ms]

[DEBUG] polls: Loading polls in navigation 
Object { app: "polls", uid: "admin", level: 0 }
Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:6:230
[DEBUG] polls: Loading polls onMounted 
Object { app: "polls", uid: "admin", level: 0 }
Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:6:230
XHRGET
https://devserver.nc.xxx/index.php/apps/polls/polls?time=1729797613902
[HTTP/2 500  263ms]

jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. 4 globals.js:36:27
session heartbeat polling started session-heartbeat.js:83:9
jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. 4 globals.js:36:27
Download the Vue Devtools extension for a better development experience:
https://github.com/vuejs/vue-devtools vue.runtime.esm.js:8809
You are running Vue in development mode.
Make sure to turn on production mode when deploying for production.
See more tips at https://vuejs.org/guide/deployment.html vue.runtime.esm.js:8818
[DEBUG] unified-search: Search providers 
Object { app: "unified-search", uid: "admin", level: 0, providers: (8) […] }
index.mjs:45:16
[DEBUG] unified-search: Contacts 
Object { app: "unified-search", uid: "admin", level: 0, contacts: (11) […] }
index.mjs:45:16
Uncaught (in promise) 
Object { stack: "Qe@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:6:36009\nrA@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:8:1033\nN@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:8:6064\nEventHandlerNonNull*A7</<@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:8:6133\nA7<@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:8:5463\nx_@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:10:512\n_request@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:11:1216\nrequest@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:10:1879\nGC/<@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:6:2651\ngetSession@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:11:21573\nload@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:1010:364332\nre@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:3:82966\nvv@https://devserver.nc.xxx/apps-writable/polls/js/polls-main.mjs:15:36016\n@https://devserver.nc.xxx/apps-writable/polls/js/polls-main.mjs:4029:216477\nua/</</u<@https://devserver.nc.xxx/apps-writable/polls/js/polls-main.mjs:15:15977\na@https://devserver.nc.xxx/apps-writable/polls/js/polls-main.mjs:15:15713\nua/</<@https://devserver.nc.xxx/apps-writable/polls/js/polls-main.mjs:15:15970\nua/<@https://devserver.nc.xxx/apps-writable/polls/js/polls-main.mjs:15:15785\nrunWithContext@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:3:39316\nH@https://devserver.nc.xxx/apps-writable/polls/js/polls-main.mjs:15:21905\nwS/ze/</<@https://devserver.nc.xxx/apps-writable/polls/js/polls-main.mjs:15:25057\npromise callback*wS/ze/<@https://devserver.nc.xxx/apps-writable/polls/js/polls-main.mjs:15:25047\nze@https://devserver.nc.xxx/apps-writable/polls/js/polls-main.mjs:15:25031\nwS/Z/<@https://devserver.nc.xxx/apps-writable/polls/js/polls-main.mjs:15:22214\npromise callback*Z@https://devserver.nc.xxx/apps-writable/polls/js/polls-main.mjs:15:22136\nR@https://devserver.nc.xxx/apps-writable/polls/js/polls-main.mjs:15:21493\nC@https://devserver.nc.xxx/apps-writable/polls/js/polls-main.mjs:15:20832\ninstall@https://devserver.nc.xxx/apps-writable/polls/js/polls-main.mjs:15:24701\nuse@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:3:38706\n@https://devserver.nc.xxx/apps-writable/polls/js/polls-main.mjs:4081:10248\n", message: "Request failed with status code 500", name: "AxiosError", code: "ERR_BAD_RESPONSE", config: {…}, request: XMLHttpRequest, response: {…}, status: 500 }

[DEBUG] core: this browser is officially supported ! 🚀 
Object { app: "core", uid: "admin", level: 0 }
index.mjs:45:16
[ERROR] polls: Error loading polls 
Object { app: "polls", uid: "admin", level: 0, error: {…} }
Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:6:434
jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please ship your own. 4 globals.js:36:27
Uncaught (in promise) 
Object { stack: "Qe@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:6:36009\nrA@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:8:1033\nN@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:8:6064\nEventHandlerNonNull*A7</<@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:8:6133\nA7<@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:8:5463\nx_@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:10:512\n_request@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:11:1216\nrequest@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:10:1879\nGC/<@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:6:2651\ngetPolls@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:11:11291\nload@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:1010:361835\nre@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:3:82966\nsetup/<@https://devserver.nc.xxx/apps-writable/polls/js/polls-main.mjs:4029:175829\nvs/</<@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:3:30981\nu4@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:3:20829\nrr@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:3:20902\na3/u.__weh@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:3:30862\nby@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:3:22447\nBy@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:3:22767\npromise callback*wy@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:3:21940\nqw@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:3:22054\nHb@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:3:58388\na4/m.scheduler@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:3:54696\nGw/_.scheduler<@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:3:20175\ntrigger@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:3:5499\ngg@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:3:5841\nnotify@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:3:8257\ntrigger@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:3:8151\nset value@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:3:17367\nX@https://devserver.nc.xxx/apps-writable/polls/js/polls-main.mjs:15:23026\nwS/R/<@https://devserver.nc.xxx/apps-writable/polls/js/polls-main.mjs:15:21692\npromise callback*R@https://devserver.nc.xxx/apps-writable/polls/js/polls-main.mjs:15:21550\nC@https://devserver.nc.xxx/apps-writable/polls/js/polls-main.mjs:15:20832\ninstall@https://devserver.nc.xxx/apps-writable/polls/js/polls-main.mjs:15:24701\nuse@https://devserver.nc.xxx/apps-writable/polls/js/Spinner.vue_vue_type_style_index_0_scoped_4bfce222_lang-D-XFACB_.chunk.mjs:3:38706\n@https://devserver.nc.xxx/apps-writable/polls/js/polls-main.mjs:4081:10248\n", message: "Request failed with status code 500", name: "AxiosError", code: "ERR_BAD_RESPONSE", config: {…}, request: XMLHttpRequest, response: {…}, status: 500 }

jQuery is deprecated: The global jQuery is deprecated. It will be removed in a later versions without another warning. Please 

### Additional client environment information

_No response_

### NC version

Other/Don't know

### Other Nextcloud version

31.0.0.3

### PHP engine version

PHP 8.2

### Other PHP version

_No response_

### Database engine

MySQL

### Database Engine version or other Database

_No response_

### Which user-backends are you using?

- [x] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other/Don't know

### Add your nextcloud server log here

_No response_

### Additional environment informations

_No response_

### Configuration report

{
    "system": {
        "debug": true,
        "profiler": true,
        "apps_paths": [
            {
                "path": "\/var\/www\/html\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/www\/html\/apps-extra",
                "url": "\/apps-extra",
                "writable": false
            },
            {
                "path": "\/var\/www\/html\/apps-shared",
                "url": "\/apps-shared",
                "writable": false
            },
            {
                "path": "\/var\/www\/html\/apps-writable",
                "url": "\/apps-writable",
                "writable": true
            }
        ],
        "allow_local_remote_servers": true,
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "1025",
        "skeletondirectory": "\/skeleton",
        "setup_create_db_user": false,
        "loglevel": 2,
        "log_query": false,
        "query_log_file": "\/shared\/log\/querylog-.log",
        "query_log_file_requestid": "yes",
        "diagnostics.logging": false,
        "diagnostics.logging.threshold": 0,
        "log.condition": {
            "apps": [
                "diagnostics",
                "admin_audit"
            ]
        },
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "nextcloud",
            "localhost"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "sqlite3",
        "version": "31.0.0.3",
        "overwrite.cli.url": "http:\/\/localhost",
        "updater.release.channel": "stable",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "lookup_server": "",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "twofactor_enforced": "false",
        "twofactor_enforced_groups": [],
        "twofactor_enforced_excluded_groups": [],
        "app_install_overwrite": [
            "twofactor_email",
            "twofactor_totp",
            "twofactor_nextcloud_notification"
        ]
    }
}

### List of activated Apps

Enabled:
  - admin_audit: 1.21.0
  - cloud_federation_api: 1.14.0
  - comments: 1.21.0
  - contactsinteraction: 1.12.0
  - dashboard: 7.11.0
  - dav: 1.32.0
  - deck: 2.0.0-dev.0
  - federatedfilesharing: 1.21.0
  - federation: 1.21.0
  - files: 2.3.0
  - files_reminders: 1.4.0
  - files_sharing: 1.23.1
  - files_trashbin: 1.21.0
  - files_versions: 1.24.0
  - lookup_server_connector: 1.19.0
  - oauth2: 1.19.1
  - polls: 8.0.0-alpha8
  - provisioning_api: 1.21.0
  - settings: 1.14.0
  - sharebymail: 1.21.0
  - sharereview: 1.2.0
  - systemtags: 1.21.0
  - theming: 2.6.1
  - twofactor_backupcodes: 1.20.0
  - twofactor_nextcloud_notification: 3.9.0
  - twofactor_totp: 6.4.1
  - updatenotification: 1.21.0
  - user_status: 1.11.0
  - viewer: 4.0.0
  - weather_status: 1.11.0
  - webhook_listeners: 1.2.0
  - workflowengine: 2.13.0
Disabled:
  - encryption: 2.19.0
  - files_external: 1.23.0
  - testing: 1.21.0
  - user_ldap: 1.22.0

### Nextcloud Signing status

_No response_

### Additional Information

Discussion in Nextcloud Forum (in German): https://help.nextcloud.com/t/polls-version-8-0-0-alpha8/206973

--- a/lib/Controller/VoteController.php
+++ b/lib/Controller/VoteController.php
@@ -55,11 +55,11 @@ class VoteController extends BaseController {
 	// #[FrontpageRoute(verb: 'PUT', url: '/vote/{optionId}/set/{setTo}')]
 	public function set(int $optionId, string $setTo): JSONResponse {
 		$option = $this->optionService->get($optionId);
+
 		return $this->response(fn () => [
 			'vote' => $this->voteService->set($optionId, $setTo),
 			'poll' => $this->pollService->get($option->getPollId()),
 			'options' => $this->optionService->list($option->getPollId()),
-		
 		]);
 	}
 
@@ -73,7 +73,12 @@ class VoteController extends BaseController {
 	#[FrontpageRoute(verb: 'DELETE', url: '/poll/{pollId}/user/{userId}', postfix: 'named')]
 	#[FrontpageRoute(verb: 'DELETE', url: '/poll/{pollId}/user', postfix: 'self')]
 	public function delete(int $pollId, string $userId = ''): JSONResponse {
-		return $this->response(fn () => ['deleted' => $this->voteService->deleteUserFromPoll($pollId, $userId)]);
+		return $this->response(fn () => [
+			'deleted' => $this->voteService->deleteUserFromPoll($pollId, $userId),
+			'poll' => $this->pollService->get($pollId),
+			'options' => $this->optionService->list($pollId),
+			'votes' => $this->voteService->list($pollId)
+		]);
 	}
 
 	/**
@@ -85,6 +90,11 @@ class VoteController extends BaseController {
 	#[OpenAPI(OpenAPI::SCOPE_IGNORE)]
 	#[FrontpageRoute(verb: 'DELETE', url: '/poll/{pollId}/votes/orphaned')]
 	public function deleteOrphaned(int $pollId, string $userId = ''): JSONResponse {
-		return $this->response(fn () => ['deleted' => $this->voteService->deleteUserFromPoll($pollId, $userId, deleteOnlyOrphaned: true)]);
+		return $this->response(fn () => [
+			'deleted' => $this->voteService->deleteUserFromPoll($pollId, $userId, deleteOnlyOrphaned: true),
+			'poll' => $this->pollService->get($pollId),
+			'options' => $this->optionService->list($pollId),
+			'votes' => $this->voteService->list($pollId)
+		]);
 	}
 }

--- a/lib/Service/VoteService.php
+++ b/lib/Service/VoteService.php
@@ -15,6 +15,7 @@ use OCA\Polls\Db\PollMapper;
 use OCA\Polls\Db\Vote;
 use OCA\Polls\Db\VoteMapper;
 use OCA\Polls\Event\VoteSetEvent;
+use OCA\Polls\Exceptions\NotFoundException;
 use OCA\Polls\Exceptions\VoteLimitExceededException;
 use OCA\Polls\UserSession;
 use OCP\AppFramework\Db\DoesNotExistException;
@@ -65,6 +66,18 @@ class VoteService {
 		return;
 	}
 
+	private function checkVoteLimit(Option $option): void {
+		// check, if the optionlimit is reached or exceeded, if one is set
+		if ($option->getIsLockedByOptionLimit()) {
+			throw new VoteLimitExceededException();
+		}
+
+		if ($option->getIsLockedByVotesLimit()) {
+			throw new VoteLimitExceededException;
+		}
+		return;
+	}
+
 	/**
 	 * Set vote
 	 */
@@ -72,6 +85,11 @@ class VoteService {
 		$option = $this->optionMapper->find($optionId);
 		$poll = $this->pollMapper->find($option->getPollId());
 		$poll->request(Poll::PERMISSION_VOTE_EDIT);
+
+		if ($option->getIsLocked()) {
+			$this->checkVoteLimit($option);
+			throw new NotFoundException();
+		}
 
 		try {
 			$this->vote = $this->voteMapper->findSingleVote($poll->getId(), $option->getPollOptionText(), $this->userSession->getCurrentUserId());

--- a/src/components/VoteTable/VoteItem.vue
+++ b/src/components/VoteTable/VoteItem.vue
@@ -67,13 +67,21 @@
 	/**
 	 *
 	 */
-	function setVote() {
+	async function setVote() {
 		if (isVotable.value) {
-			votesStore.set({
-				option: props.option,
-				setTo: nextAnswer.value,
-			})
-			showSuccess(t('polls', 'Vote saved'), { timeout: 2000 })
+			try {
+				await votesStore.set({
+					option: props.option,
+					setTo: nextAnswer.value,
+				})
+				showSuccess(t('polls', 'Vote saved'), { timeout: 2000 })
+			} catch (error) {
+				if (error.response.status === 409 && error.response.data.message === 'Vote limit exceeded') {
+					showError(t('polls', 'Vote already booked out'))
+				} else {
+					showError(t('polls', 'Error saving vote'))
+				}
+			}
 		} else {
 			showError(t('polls', 'Error saving vote'))
 		}

--- a/src/stores/votes.ts
+++ b/src/stores/votes.ts
@@ -81,7 +81,7 @@ export const useVotesStore = defineStore('votes', {
 					this.$reset()
 					return
 				}
-	
+
 				const votes: Vote[] = []
 				response.data.votes.forEach((vote: Vote) => {
 					if (vote.answer === Answer.Yes) {
@@ -104,7 +104,7 @@ export const useVotesStore = defineStore('votes', {
 				throw error
 			}
 		},
-	
+
 		setItem(payload: { option: Option; vote: Vote }) {
 			const index = this.list.findIndex((vote: Vote) =>
 				vote.pollId === payload.option.pollId
@@ -116,7 +116,7 @@ export const useVotesStore = defineStore('votes', {
 			}
 			this.list.push(payload.vote)
 		},
-	
+
 		async set(payload: { option: Option; setTo: Answer }) {
 			const sessionStore = useSessionStore()
 			const optionsStore = useOptionsStore()
@@ -135,17 +135,17 @@ export const useVotesStore = defineStore('votes', {
 			} catch (error) {
 				if (error?.code === 'ERR_CANCELED') return
 				if (error.response.status === 409) {
-					this.load()
-					optionsStore.load()
 					pollStore.load()
+					throw error
 				} else {
-					Logger.error('Error setting vote', { error, payload })
+					Logger.error('Error setting vote aa', { error, payload })
 					throw error
 				}
 			}
 		},
-	
+
 		async resetVotes() {
+			Logger.debug('Resetting votes')
 			const sessionStore = useSessionStore()
 			try {
 				let response = null
@@ -162,7 +162,7 @@ export const useVotesStore = defineStore('votes', {
 				throw error
 			}
 		},
-	
+
 		async deleteUser(payload) {
 			const sessionStore = useSessionStore()
 			try {
@@ -178,7 +178,6 @@ export const useVotesStore = defineStore('votes', {
 		async removeOrphanedVotes() {
 			const sessionStore = useSessionStore()
 			const pollStore = usePollStore()
-			const optionsStore = useOptionsStore()
 			try {
 				if (sessionStore.route.name === 'publicVote') {
 					await PublicAPI.removeOrphanedVotes(sessionStore.route.params.token)
@@ -186,13 +185,12 @@ export const useVotesStore = defineStore('votes', {
 					await VotesAPI.removeOrphanedVotes(sessionStore.route.params.id)
 				}
 				pollStore.load()
-				optionsStore.load()
 			} catch (error) {
 				if (error?.code === 'ERR_CANCELED') return
 				Logger.error('Error deleting orphaned votes', { error })
 				throw error
 			}
 		},
-	
+
 	},
 })

